### PR TITLE
[RLlib] A2C + A3C move to `algorithms` folder and re-name into A2C/A3C (from ...Trainer).

### PR DESCRIPTION
**Pull request by sven1977**
A2C + A3C:
* move Trainers to `algorithms` folder.
* re-name Trainers into `A2C/A3C` (from `A2C/A3CTrainer`).
* separate them in different directories (`rllib/algorithms/a3c` and `rllib/algorithms/a2c`)

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/a3c/README.md [hunk 1]
```diff
@@ -1,22 +0,0 @@
-# Advantage Actor-Critic (A2C, A3C)
-
-## Overview 
-
-[Advantage Actor-Critic](https://arxiv.org/pdf/1602.01783.pdf) proposes two distributed model-free on-policy RL algorithms, A3C and A2C. These algorithms are distributed versions of the vanilla Policy Gradient (PG) algorithm with different distributed execution patterns. The paper suggests accelerating training via scaling data collection, i.e. introducing worker nodes, which carry copies of the central node's policy network and collect data from the environment in parallel. This data is used on each worker to compute gradients. The central node applies each of these gradients and then sends updated weights back to the workers.
```
Comment (maxpumperla) [inline rllib/agents/a3c/README.md:5]: why remove this? is this duplicated in docs?

File: rllib/algorithms/a2c/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Advantage Actor-Critic (A2C)
+
+## Overview 
+
+[Advantage Actor-Critic](https://arxiv.org/pdf/1602.01783.pdf) proposes two distributed model-free on-policy RL algorithms, A3C and A2C.
+These algorithms are distributed versions of the vanilla Policy Gradient (PG) algorithm with different distributed execution patterns.
+The paper suggests accelerating training via scaling data collection, i.e. introducing worker nodes,
+which carry copies of the central node's policy network and collect data from the environment in parallel.
```
Comment (maxpumperla) [inline rllib/algorithms/a2c/README.md:8]: ok, got it :D

**Review by maxpumperla — APPROVED** [review]
(ignore comments, all good)

File: rllib/agents/a3c/README.md (hunk 1 above)
Comment (sven1977) [reply rllib/agents/a3c/README.md:5]: Yeah, I separated A2C and A3C as discussed in our doc.

**Pull request merged**